test/middleware/LoadAppealMiddleware.test.ts

```diff
@@ -0,0 +1,122 @@
+import 'reflect-metadata';
+
+import Substitute, { Arg } from '@fluffy-spoon/substitute';
+import { Maybe } from 'ch-node-session-handler';
+import { SessionKey } from 'ch-node-session-handler/lib/session/keys/SessionKey';
+import { expect } from 'chai';
+import { NextFunction, Request, Response } from 'express';
+import { createSubstituteOf } from '../SubstituteFactory';
+import { APPEAL_WITH_ATTACHMENTS } from '../models/AppDataFactory';
+import { createSession } from '../utils/session/SessionFactory';
+
+import { APPEAL_ID_QUERY_KEY, COMPANY_NUMBER_QUERY_KEY, LoadAppealMiddleware } from 'app/middleware/LoadAppealMiddleware';
+import { Appeal } from 'app/models/Appeal';
+import { ApplicationData, APPLICATION_DATA_KEY } from 'app/models/ApplicationData';
+import { AppealsService } from 'app/modules/appeals-service/AppealsService';
+import { AppealNotFoundError, AppealServiceError } from 'app/modules/appeals-service/errors';
+
+describe('LoadAppealMiddleware', () => {
+
+    const appealId = '123';
+    const companyId = 'NI000000';
+    // @ts-ignore
+    const createAppealService = (method: 'resolves' | 'rejects', data?: Appeal | any) => {
```

Comment:
I do not see much value in having this method since there is `createSubstituteOf` ready for use as:

```
createSubstituteOf<Response>(service => {
    service.getAppeal(Arg.all()).resolves(data);
})
```

Reply:
sure I can use the `createSubstituteOf` method but at the end of the day I believe it saves a few lines and makes the code more readable.

```
const createAppealService = (method: 'resolves' | 'rejects', data?: Appeal | any) =>
        createSubstituteOf<AppealsService>(service => service.getAppeal(Arg.all())[method](data));

const service = createAppealService('resolves', something); // x4 vs.

const service = createSubstituteOf<AppealsService>(service => service.getAppeal(Arg.all()).resolves(something)); // x4
```

Reply:
General speaking number of lines is not decisive factor at all. Readability and maintainability is much more important. Having 5, 10, 15, 50 lines does not matter much to me. I look for the code which is simple, easy to read / follow and change.